### PR TITLE
Javascript/javascript2/week3

### PR DESCRIPTION
**Pull request by shpomp**

### CODE REVIEW

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+window.ConfettiGenerator = function (e) {
+  var t = {
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:2]:
```suggestion
  const t = {
```

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    (e.target && (t.target = e.target),
+    e.max && (t.max = e.max),
+    e.size && (t.size = e.size),
+    void 0 !== e.animate && null !== e.animate && (t.animate = e.animate),
+    e.props && (t.props = e.props),
+    e.colors && (t.colors = e.colors),
+    e.clock && (t.clock = e.clock),
+    e.width && (t.width = e.width),
+    e.height && (t.height = e.height));
+  var i = document.getElementById(t.target),
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:32]:
```suggestion
  let i = document.getElementById(t.target),
```

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    a = i.getContext("2d"),
+    r = [];
+  function o(e, t) {
+    e || (e = 1);
+    var i = Math.random() * e;
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:37]:
```suggestion
    let i = Math.random() * e;
```

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    return t ? Math.floor(i) : i;
+  }
+  function n(e) {
+    var i = e.radius <= 3 ? 0.4 : 0.8;
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:41]:
```suggestion
    let i = e.radius <= 3 ? 0.4 : 0.8;
```

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          angles: [o(10, !0) + 2, o(10, !0) + 2, o(10, !0) + 2, o(10, !0) + 2],
+          color: t.colors[o(t.colors.length, !0)],
+          rotation: (o(360, !0) * Math.PI) / 180,
+          speed: o(t.clock / 7) + t.clock / 30,
+        });
+      return (function e() {
+        for (var i in (a.clearRect(0, 0, t.width, t.height), r)) n(r[i]);
+        !(function () {
+          for (var e = 0; e < t.max; e++) {
+            var i = r[e];
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:92]:
```suggestion
            let i = r[e];
```

File: javascript/javascript2/week3/fastestPresser/confetti.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        })(),
+          t.animate && requestAnimationFrame(e);
+      })();
+    },
+    clear: function () {
+      (t.animate = !1),
+        clearInterval(t.interval),
+        requestAnimationFrame(function () {
+          a.clearRect(0, 0, i.width, i.height);
+          var e = i.width;
```
Comment (valmdz) [inline javascript/javascript2/week3/fastestPresser/confetti.js:106]:
```suggestion
          let e = i.width;
```

File: javascript/javascript2/week3/geo/index.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+function geoFindMe() {
+    const locationStatus = document.querySelector("#locationStatus");
+    function success(position) {
+        const latitude  = position.coords.latitude;
+        const longitude = position.coords.longitude;
+      locationStatus.innerHTML = `Latitude: ${latitude} °, Longitude: ${longitude} °`;
+    
+    // the google map is not really working, even through I generated a key and tried.
```
Comment (valmdz) [inline javascript/javascript2/week3/geo/index.js:8]: I remember I also tried showing the map but was not successful at the time, have not tried again either

File: javascript/javascript2/week3/inclass.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+console.log("script");
+
+//Click counter
+//When the button first is clicked it should first log out 0. The next time it is clicked it should log out 1, etc.
+//Clicking the second button should also count up and logout the same variable.
+
+let button1 = document.getElementById("button1");
```
Comment (valmdz) [inline javascript/javascript2/week3/inclass.js:7]:
```suggestion
const button1 = document.getElementById("button1");
```

File: javascript/javascript2/week3/inclass.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+let button2 = document.getElementById("button2");
```
Comment (valmdz) [inline javascript/javascript2/week3/inclass.js:8]:
```suggestion
const button2 = document.getElementById("button2");
```

File: javascript/javascript2/week3/warm-up.js [hunk 10]
```diff
@@ -0,0 +1,153 @@
+console.log("script loaded");
+
+//1. Log out the text "Called after 2.5 seconds" 2.5 seconds after the script is loaded.
+let miliseconds = 2500;
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:4]:
```suggestion
const miliseconds = 2500;
```

File: javascript/javascript2/week3/warm-up.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+let seconds = miliseconds / 1000;
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:5]:
```suggestion
const seconds = miliseconds / 1000;
```

File: javascript/javascript2/week3/warm-up.js [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+let after2point5seconds = setTimeout(
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:7]:
```suggestion
const after2point5seconds = setTimeout(
```

File: javascript/javascript2/week3/warm-up.js [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+  miliseconds
+);
+
+//console.log(after2point5seconds);
+
+//2. Create a function that takes 2 parameters: delay and stringToLog.
+// Calling this function should lo g out the stringToLog after delay seconds.
+// Call the function you have created with some different arguments.
+
+let stringToLog = "what a beautiful day!";
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:18]:
```suggestion
const stringToLog = "what a beautiful day!";
```

File: javascript/javascript2/week3/warm-up.js [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+delayedString(delay, stringToLog);
+// const callDelayed = () => delayedString(delay, stringToLog);
+
+//----3. Create a button in html. When clicking this button,
+// use the function you created in the previous task to log out the text:
+// "Called after 5 seconds", 5 seconds after the button is clicked.
+const newDelay = 5;
+const newStringToLog = "Logged after 5 seconds";
+
+let buttonClick = document.createElement("button");
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:36]:
```suggestion
const buttonClick = document.createElement("button");
```

File: javascript/javascript2/week3/warm-up.js [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+// 7. Create a function called runAfterDelay.
+// It has two parameters: delay and callback.
+// When called the function should wait delay seconds and then call the provided callback function.
+// Try and call this function with different delays and different callback functions
+
+function runAfterDelay(delay1, callback) {
+  setTimeout(callback, delay1 * 1000);
+}
+
+let delay2 = 7;
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:76]:
```suggestion
const delay2 = 7;
```

File: javascript/javascript2/week3/warm-up.js [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+runAfterDelay(delay2, function () {
+  console.log(`Logged after 7 seconds`);
+});
+
+let delay3 = 9;
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:81]:
```suggestion
const delay3 = 9;
```

File: javascript/javascript2/week3/warm-up.js [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+  appearingButton.innerText = `I appeared after ${delay3} seconds!`;
+  document.body.appendChild(appearingButton);
+});
+
+// 8. Check if we have double clicked on the page.
+// A double click is defined by two clicks within 0.5 seconds.
+// If a double click has been detected, log out the text: "double click!"
+
+function doubleclicked() {
+  let dblclck = document.createElement("p");
```
Comment (valmdz) [inline javascript/javascript2/week3/warm-up.js:93]:
```suggestion
  const dblclck = document.createElement("p");
```

**Review by valmdz — COMMENTED** [review]
@shpomp I think all looks pretty good, just be careful when using `let `and `const`, and try to not use `var `;) As an extra suggestion, try running a formatter to remove extra spaces in the code. 
Super well done 👍

**Comment by shpomp** [comment]
Commited change suggestions

**Pull request merged**